Tests/SotoCoreTests/Credential/ConfigFileCredentialProviderTests.swift

```diff
@@ -20,171 +20,93 @@ import SotoXML
-        let credential = """
-        [\(profile)]
-        aws_secret_access_key=\(secretKey)
-        """
-
-        var byteBuffer = ByteBufferAllocator().buffer(capacity: credential.utf8.count)
-        byteBuffer.writeString(credential)
-        XCTAssertThrowsError(_ = try AWSConfigFileCredentialProvider.sharedCredentials(from: byteBuffer, for: profile)) {
-            XCTAssertEqual($0 as? AWSConfigFileCredentialProvider.ConfigFileError, .missingAccessKeyId)
+    func makeContext() throws -> (CredentialProviderFactory.Context, MultiThreadedEventLoopGroup, HTTPClient) {
+        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        let eventLoop = eventLoopGroup.next()
+        let httpClient = HTTPClient(eventLoopGroupProvider: .shared(eventLoop))
+        return (.init(httpClient: httpClient, eventLoop: eventLoop, logger: TestEnvironment.logger), eventLoopGroup, httpClient)
+    }
+
+    func testCredentialProvider() throws {
+        let credentials = ConfigFileLoader.ProfileCredentials(
+            accessKey: "foo",
+            secretAccessKey: "bar",
+            sessionToken: nil,
+            roleArn: nil,
+            roleSessionName: nil,
+            sourceProfile: nil,
+            credentialSource: nil)
+        let (context, eventLoopGroup, httpClient) = try makeContext()
+
+        defer {
+            try? httpClient.syncShutdown()
+            try? eventLoopGroup.syncShutdownGracefully()
         }
-    }
-
-    func testConfigFileCredentialsMissingSecretKey() {
-        let profile = "profile1"
-        let accessKey = "FAKE-ACCESS-KEY123"
-        let credential = """
-        [\(profile)]
-        aws_access_key_id=\(accessKey)
-        """
 
-        var byteBuffer = ByteBufferAllocator().buffer(capacity: credential.utf8.count)
-        byteBuffer.writeString(credential)
-        XCTAssertThrowsError(_ = try AWSConfigFileCredentialProvider.sharedCredentials(from: byteBuffer, for: profile)) {
-            XCTAssertEqual($0 as? AWSConfigFileCredentialProvider.ConfigFileError, .missingSecretAccessKey)
+        let provider = try ConfigFileCredentialProvider.credentialProvider(
+            from: credentials,
+            config: nil,
+            context: context
+        )
+        XCTAssertEqual((provider as? StaticCredential)?.accessKeyId, "foo")
+        XCTAssertEqual((provider as? StaticCredential)?.secretAccessKey, "bar")
+    }
+
+    func testCredentialProviderSTSAssumeRole() throws {
+        let credentials = ConfigFileLoader.ProfileCredentials(
+            accessKey: "foo",
+            secretAccessKey: "bar",
+            sessionToken: nil,
+            roleArn: "arn",
+            roleSessionName: nil,
+            sourceProfile: "baz",
+            credentialSource: nil)
+        let (context, eventLoopGroup, httpClient) = try makeContext()
+
+        defer {
+            try? httpClient.syncShutdown()
+            try? eventLoopGroup.syncShutdownGracefully()
```

Comment:
You need to call `shutdown` on the credential provider in this test and `wait()` on it finishing